### PR TITLE
[SW-1628] Create migration guide document with breaking changes so far added to 3.28

### PR DESCRIPTION
**Pull request by jakubhava**
Migration guide between major SW versions. I have included changes which goes to next major 3.28 and are breaking the code. It would be great if every change going to major rel which is altering the current API has also short description in this document

### CODE REVIEW

File: doc/src/site/sphinx/migration_guide.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+===============
+
+Migration guide between major Sparkling Water versions
+
+From 3.26 To 3.28
+-----------------
+
+String instead of enums in Sparkling Water Algo API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- In scala, Setters on the pipeline wrappers for H2O algorithms now accepts strings in places where they accepted
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `Setters on` -> `setters of`?

File: doc/src/site/sphinx/migration_guide.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Now, the correct code is:
+
+.. code-block:: scala
+
+    val gbm = H2OGBM()
+    gbm.setDistribution("multinomial")
+
+which makes the Python and Scala consistent. Both upper case and lower case values are valid and if a wrong
+input is entered, user is warned with correct possible values.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `user` -> `a user`?

File: doc/src/site/sphinx/migration_guide.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+DRF exposed into Sparkling Water Algorithm API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+DRF is now exposed in the Sparkling Water. Please see our documentation to learn how to use it :ref:`drf`.
+
+Single value in prediction column
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The default name of the prediction column is changed to ``prediction`` from ``prediction_output``. Also, the
+prediction output now contains always directly the predicted value. In case of regression issue the predicted numeric value
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `Also, the
prediction output now contains always directly the predicted value. In case of regression issue the predicted numeric value and in case of classification the predicted label.` Can you rephrase it?

File: doc/src/site/sphinx/migration_guide.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+probabilities, contributions and so on.
+
+Removal of Deprecated Methods and Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- ``getColsampleBytree and setColsampleBytree`` methods are removed on our algorithm API. Please use
+  the new ``getColSampleByTree`` and ``setColSampleByTree``
+
+- Remove deprecated option ``spark.ext.h2o.external.cluster.num.h2o.nodes`` and corresponding setters.
+  Please use ``spark.ext.h2o.external.cluster.size`` or corresponding setter ``setClusterSize``
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `corresponding setter` -> `the corresponding setter`?

File: doc/src/site/sphinx/migration_guide.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+When enabled, additional column ``detailed_prediction`` is created which contains additional prediction details, such as
+probabilities, contributions and so on.
+
+Removal of Deprecated Methods and Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- ``getColsampleBytree and setColsampleBytree`` methods are removed on our algorithm API. Please use
+  the new ``getColSampleByTree`` and ``setColSampleByTree``
+
+- Remove deprecated option ``spark.ext.h2o.external.cluster.num.h2o.nodes`` and corresponding setters.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: shouldn't the the points be written in past tense? `Remove` -> `Removed`?

File: doc/src/site/sphinx/migration_guide.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+- Deprecated Algorithm classes in package ``org.apache.spark.h2o.ml.algos`` have been removed. Please
+  use the classes from the package ``ai.h2o.sparkling.ml.algos``. Their API remains the same. This is the
+  beginning of moving Sparkling Water classes to our distinct package.
+
+- Remove deprecated option ``spark.ext.h2o.external.read.confirmation.timeout`` and related setters.
+  This option is removed without replacement as it is no longer needed.
+
+- Remove deprecated parameter ``SelectBestModelDecreasing`` on our Grid Search API. Related getters and setters
+  have been also removed. This method is removed without replacement as we now internally correctly sort
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `replacement` ->  `a replacement`?

File: doc/src/site/sphinx/migration_guide.rst [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  the models according to what is best for the specified metric.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `best` -> `convenient`?

Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `the specified metric` -> `the specified sort metric`?

File: doc/src/site/sphinx/migration_guide.rst [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+- TargetEncoder transformer now accepts outputCols parameter which you can use to override the default output
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `outputCols parameter` ->  `the ``outputCols`` parameter`?

File: doc/src/site/sphinx/migration_guide.rst (hunk 1 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: 👍

File: doc/src/site/sphinx/migration_guide.rst (hunk 2 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: Rephrased a bit as well, thanks!

File: doc/src/site/sphinx/migration_guide.rst (hunk 8 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: 👍

File: doc/src/site/sphinx/migration_guide.rst (hunk 7 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: 👍

Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: Rephrased this a bit as well

File: doc/src/site/sphinx/migration_guide.rst (hunk 5 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: Agree. In this case I would start with `Removal of`, what do you think? But otherwise I would also prefer using past tense.

File: doc/src/site/sphinx/migration_guide.rst (hunk 6 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: 👍

File: doc/src/site/sphinx/migration_guide.rst (hunk 4 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: 👍

File: doc/src/site/sphinx/migration_guide.rst (hunk 3 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: Rephrased, please check if it make more sense now

File: doc/src/site/sphinx/migration_guide.rst [hunk 9]
```diff
@@ -38,14 +38,19 @@ DRF exposed into Sparkling Water Algorithm API
 
 DRF is now exposed in the Sparkling Water. Please see our documentation to learn how to use it :ref:`drf`.
 
+Change Default Name of Prediction Column
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The default name of the prediction column is changed to ``prediction`` from ``prediction_output``.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: nit: `The default name of the prediction column has been changed from ``prediction_output`` to ``prediction`` `?

File: doc/src/site/sphinx/migration_guide.rst [hunk 10]
```diff
@@ -38,14 +38,19 @@ DRF exposed into Sparkling Water Algorithm API
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The default name of the prediction column is changed to ``prediction`` from ``prediction_output``. Also, the
-prediction output now contains always directly the predicted value. In case of regression issue the predicted numeric value
-and in case of classification the predicted label. If you are interested in more details created during the prediction,
-please make sure to set ``withDetailedPredictionCol`` to ``true`` via the setters on both PySparkling Scala side.
-When enabled, additional column ``detailed_prediction`` is created which contains additional prediction details, such as
+The prediction column contains directly the predicted value. For example, before this change, the prediction column contained
+another struct field called ``value`` (in case of regression issue), which contained the value. From now on, the predicted value
+is always stored directly in the prediction column. In case of regression issue,  the predicted numeric value
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: nit: ` issue,  the` extra space

File: doc/src/site/sphinx/migration_guide.rst [hunk 11]
```diff
@@ -51,22 +51,22 @@ probabilities, contributions and so on.
...
 
-- Remove deprecated option ``spark.ext.h2o.external.cluster.num.h2o.nodes`` and corresponding setters.
-  Please use ``spark.ext.h2o.external.cluster.size`` or corresponding setter ``setClusterSize``
+- Removal of deprecated option ``spark.ext.h2o.external.cluster.num.h2o.nodes`` and corresponding setters.
+  Please use ``spark.ext.h2o.external.cluster.size`` or the corresponding setter ``setClusterSize``.
 
-- Deprecated Algorithm classes in package ``org.apache.spark.h2o.ml.algos`` have been removed. Please
-  use the classes from the package ``ai.h2o.sparkling.ml.algos``. Their API remains the same. This is the
-  beginning of moving Sparkling Water classes to our distinct package.
+- Removal of of deprecated algorithm classes in package ``org.apache.spark.h2o.ml.algos``. Please
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: `of` is there twice

Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: great catch, thanks!

File: doc/src/site/sphinx/migration_guide.rst (hunk 10 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: thanks!

File: doc/src/site/sphinx/migration_guide.rst (hunk 9 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: Yul, like it better!

File: doc/src/site/sphinx/migration_guide.rst [hunk 12]
```diff
@@ -51,22 +51,22 @@ probabilities, contributions and so on.
...
-  This option is removed without replacement as it is no longer needed.
+- Removal of deprecated option ``spark.ext.h2o.external.read.confirmation.timeout`` and related setters.
+  This option is removed without a replacement as it is no longer needed.
 
-- Remove deprecated parameter ``SelectBestModelDecreasing`` on our Grid Search API. Related getters and setters
-  have been also removed. This method is removed without replacement as we now internally correctly sort
-  the models according to what is best for the specified metric.
+- Removal of deprecated parameter ``SelectBestModelDecreasing`` on the Grid Search API. Related getters and setters
+  have been also removed. This method is removed without replacement as we now internally sort
+  the models with the ordering meaningful for the specified sort metric.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: nit: maybe `meaningful to` would be better

Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: Oki, will fix in the following small change, thanks!
